compiler/quilt/tools/build.py

```diff
@@ -80,13 +81,19 @@ def _is_valid_group(group):
     return isinstance(group, dict) or group is None
 
 def _pythonize_name(name):
-    safename = re.sub('[^A-Za-z0-9]+', '_', name).strip('_')
-
-    if safename and safename[0].isdigit():
-        safename = "n%s" % safename
+    """Convert `name` to a Python Identifier, but raise BuildException on failure."""
+    try:
+        safename = to_identifier(name, strip_underscores=True)
+    except ValueError as error:
+        raise BuildException("Invalid name: " + str(error))
+    return safename
 
-    if not is_nodename(safename):
-        raise BuildException("Unable to determine a Python-legal name for %r" % name)
+def to_nodename(name, invalid=None):
```

Comment:
I'm worried that having both `util.to_nodename` and `build.to_nodename` is going to create confusion. I'd make this one "private" - i.e., call it `_to_nodename`, or even something more obscure.

Reply:
`_to_nodename` is what I did originally, but I got called out by Kevin for doing it the other way, because it then involved renaming existing usages -- but, that was in the other (larger) PR, and I think it was just TMI there.  I'll switch it to `to_nodename_reraise` or something similar.